test/wallet-test.js

```diff
@@ -2279,4 +2289,115 @@ describe('Wallet', function() {
+        const cur = new ChainEntry(curBlock(wdb));
+        const next = new Block(nextBlock(wdb));
+        const entry = ChainEntry.fromBlock(next, cur);
+        await wdb.addBlock(entry, []);
+      }
+
+      const bal = await wallet.getBalance();
+      assert.equal(bal.tx, 0);
+      assert.equal(bal.coin, 0);
+      assert.equal(bal.unconfirmed, 0);
+      assert.equal(bal.confirmed, 0);
+      assert.equal(bal.ulocked, 0);
+      assert.equal(bal.clocked, 0);
+    });
+
+    after(async () => {
+      await wdb.close();
+    });
+
+    it('should not have any cloudflare state', async () => {
+      const rawName = Buffer.from('cloudflare', 'ascii');
+      const nameHash = rules.hashName(rawName);
+      const nameinfo = await wallet.getNameState(nameHash);
+      assert.deepEqual(nameinfo, null);
+    });
+
+    it('should confirm cloudflare CLAIM', async () => {
+      const claim = await wallet.sendFakeClaim('cloudflare');
+      assert(claim);
+
+      const tx = claim.toTX(network, wdb.state.height + 1);
+
+      const cur = new ChainEntry(curBlock(wdb));
+      const next = new Block(nextBlock(wdb));
+      const entry = ChainEntry.fromBlock(next, cur);
+      await wdb.addBlock(entry, [tx]);
+
+      const rawName = Buffer.from('cloudflare', 'ascii');
+      const nameHash = rules.hashName(rawName);
+      const ns = await wallet.getNameState(nameHash);
+
+      const json = ns.getJSON(wdb.state.height, network);
+      assert.equal(json.name, 'cloudflare');
+      assert.equal(json.state, 'LOCKED');
+
+      const bal = await wallet.getBalance();
+      assert.equal(bal.tx, 1);
+      assert.equal(bal.coin, 1);
+
+      // TODO: balance tests
+    });
+
+    it('should close the auction', async () => {
+      const rawName = Buffer.from('cloudflare', 'ascii');
+      const nameHash = rules.hashName(rawName);
+      const ns = await wallet.getNameState(nameHash);
+      const json = ns.getJSON(wdb.state.height, network);
+      const {blocksUntilClosed} = json.stats;
+
+      for (let i = 0; i < blocksUntilClosed; i++) {
+        const cur = new ChainEntry(curBlock(wdb));
+        const next = new Block(nextBlock(wdb));
+        const entry = ChainEntry.fromBlock(next, cur);
+        await wdb.addBlock(entry, []);
+      }
+
+      {
+        const ns = await wallet.getNameState(nameHash);
+        const json = ns.getJSON(wdb.state.height, network);
+        assert.equal(json.name, 'cloudflare');
+        assert.equal(json.state, 'CLOSED');
+      }
+    });
+
+    it('should send an update for cloudflare', async () => {
+      const records = Resource.fromJSON({
+        records: [{type: 'NS', ns: 'ns1.easyhandshake.com.'}]
+      });
+
+      const update = await wallet.sendUpdate('cloudflare', records);
+
+      // Confirm UPDATE
+      const cur = new ChainEntry(curBlock(wdb));
+      const next = new Block(nextBlock(wdb));
+      const entry = ChainEntry.fromBlock(next, cur);
+      await wdb.addBlock(entry, [update]);
```

Comment:
This sequence comes up a few times as well, maybe could be replaced with a helper function at the top of the `describe` block.